samples/snippets/src/main/java/com/example/spanner/PgSpannerSample.java

```diff
@@ -0,0 +1,1603 @@
+      throw (SpannerException) e.getCause();
+    } catch (InterruptedException e) {
+      // Throw when a thread is waiting, sleeping, or otherwise occupied,
+      // and the thread is interrupted, either before or during the activity.
+      throw SpannerExceptionFactory.propagateInterrupt(e);
+    }
+  }
+  // [END spanner_create_table_using_ddl]
+
+  // [START spanner_read_stale_data]
+  static void readStaleData(DatabaseClient dbClient) {
+    try (ResultSet resultSet =
+             dbClient
+                 .singleUse(TimestampBound.ofExactStaleness(15, TimeUnit.SECONDS))
+                 .read(
+                     "Albums", KeySet.all(),
+                     Arrays.asList("SingerId", "AlbumId", "MarketingBudget"))) {
+      while (resultSet.next()) {
+        System.out.printf(
+            "%d %d %s\n",
+            resultSet.getLong(0),
+            resultSet.getLong(1),
+            resultSet.isNull(2) ? "NULL" : resultSet.getLong(2));
+      }
+    }
+  }
+  // [END spanner_read_stale_data]
+
+  // Before executing this method, a new column MarketingBudget has to be added to the Albums
+  // table by applying the DDL statement "ALTER TABLE Albums ADD COLUMN MarketingBudget BIGINT".
+  // In addition this update expects the LastUpdateTime column added by applying the DDL statement
+  // "ALTER TABLE Albums ADD COLUMN LastUpdateTime TIMESTAMPTZ"
+  // [START spanner_update_data_with_timestamp_column]
+  static void updateWithTimestamp(DatabaseClient dbClient) {
+    // Mutation can be used to update/insert/delete a single row in a table. Here we use
+    // newUpdateBuilder to create update mutations.
+    List<Mutation> mutations =
+        Arrays.asList(
+            Mutation.newUpdateBuilder("Albums")
+                .set("SingerId")
+                .to(1)
+                .set("AlbumId")
+                .to(1)
+                .set("MarketingBudget")
+                .to(1000000)
+                .set("LastUpdateTime")
+                .to(Timestamp.now())
+                .build(),
+            Mutation.newUpdateBuilder("Albums")
+                .set("SingerId")
+                .to(2)
+                .set("AlbumId")
+                .to(2)
+                .set("MarketingBudget")
+                .to(750000)
+                .set("LastUpdateTime")
+                .to(Timestamp.now())
+                .build());
+    // This writes all the mutations to Cloud Spanner atomically.
+    dbClient.write(mutations);
+  }
+  // [END spanner_update_data_with_timestamp_column]
+
+  // [START spanner_add_timestamp_column]
+  static void addLastUpdateTimestampColumn(DatabaseAdminClient adminClient, DatabaseId dbId) {
+    OperationFuture<Void, UpdateDatabaseDdlMetadata> op =
+        adminClient.updateDatabaseDdl(
+            dbId.getInstanceId().getInstance(),
+            dbId.getDatabase(),
+            Arrays.asList(
+                "ALTER TABLE Albums ADD COLUMN LastUpdateTime TIMESTAMPTZ"),
```

Comment:
Are commit timestamps not (yet) supported?

Reply:
I tried, it is not supported and couldn't find doc as well for commit timestamp for PG.

Reply:
use spanner.commit_timestamp as the type when adding column